# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/OriginInfo.h
# Commit: d63dac1ee4d4
# Full Hash: d63dac1ee4d471defff484dae5a3535e1f8ba072
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 22:03:45
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Make it possible to initialize temporary origins without ensuring origin directories; r=dom-storage-reviewers,jari
#   
#   LSNG already uses some QuotaManager APIs to achieve that origin directories are
#   not created if they don't exist during datastore preparation, but the feature
#   is not easy to use and it's also not generalized enough for use in other quota
# ==============================================================================

diff -r 2ab08c49f2b1 -r d63dac1ee4d4 dom/quota/OriginInfo.h
--- a/dom/quota/OriginInfo.h	Thu Oct 10 12:22:54 2024 +0000
+++ b/dom/quota/OriginInfo.h	Thu Oct 10 12:22:55 2024 +0000
@@ -19,6 +19,7 @@
 class OriginInfo final {
   friend class CanonicalQuotaObject;
   friend class GroupInfo;
+  friend class PersistOp;
   friend class QuotaManager;
 
  public:
@@ -65,6 +66,12 @@
 
   bool IsExtensionOrigin() const { return mIsExtension; }
 
+  bool LockedDirectoryExists() const {
+    AssertCurrentThreadOwnsQuotaMutex();
+
+    return mDirectoryExists;
+  }
+
   OriginMetadata FlattenToOriginMetadata() const;
 
   FullOriginMetadata LockedFlattenToFullOriginMetadata() const;
@@ -96,6 +103,8 @@
 
   void LockedPersist();
 
+  void LockedDirectoryCreated();
+
   nsTHashMap<nsStringHashKey, NotNull<CanonicalQuotaObject*>>
       mCanonicalQuotaObjects;
   ClientUsageArray mClientUsages;
@@ -114,7 +123,7 @@
    * want to be able to track quota for an origin without creating its origin
    * directory or the per-client files until they are actually needed to store
    * data. In those cases, the OriginInfo will be created by
-   * EnsureQuotaForOrigin and the resulting mDirectoryExists will be false until
+   * InitQuotaForOrigin and the resulting mDirectoryExists will be false until
    * the origin actually needs to be created. It is possible for mUsage to be
    * greater than zero while mDirectoryExists is false, representing a state
    * where a client like LocalStorage has reserved quota for disk writes, but